packages/client/components/MyDashboardTasks.tsx

```diff
@@ -17,20 +17,18 @@ interface Props {
 
 const MyDashboardTasks = (props: Props) => {
   const {retry, viewer} = props
-  const {showArchivedTasksCheckbox, teamFilter} = viewer
   const atmosphere = useAtmosphere()
-  const {viewerId} = atmosphere
+
+  const {userIds, teamIds, showArchived} = parseUserTaskFilters(atmosphere.viewerId)
+
   useStoreQueryRetry(retry)
   useDocumentTitle('My Tasks | Parabol', 'My Tasks')
-  useEffect(() => {
-    return () => filterTeam(atmosphere, null)
-  }, [])
   return (
     <>
       <UserTasksHeader viewer={viewer} />
```

Comment:
if you promote this to a parent component then it won't flicker when you change the filter.
Alternatively, you can memoize the component, which should have the same effect